Std/Tactic/PrintPrefix.lean

```diff
@@ -4,36 +4,147 @@ Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Shing Tak Lam, Daniel Selsam, Mario Carneiro
 -/
 import Std.Lean.Util.EnvSearch
-import Lean.Elab.Command
+import Lean.Elab.Tactic.Config
 
 namespace Lean.Elab.Command
 
-private def appendMatchingConstants (msg : String)
-    (ϕ : ConstantInfo → MetaM Bool) (opts : EnvironmentSearchOptions := {}) : MetaM String := do
-  let cinfos ← getMatchingConstants ϕ opts
-  let cinfos := cinfos.qsort fun p q => p.name.lt q.name
-  let mut msg := msg
-  for cinfo in cinfos do
-    msg := msg ++ s!"{cinfo.name} : {← Meta.ppExpr cinfo.type}\n"
-  pure msg
+/--
+Options to control `#print prefix` command and `getMatchingConstants`.
+-/
+structure PrintPrefixConfig where
+  /-- Include declarations in imported environment. -/
+  imported : Bool := true
+  /-- Include declarations whose types are propositions. -/
+  propositions : Bool := true
+  /-- Exclude declarations whose types are not propositions. -/
+  propositionsOnly : Bool := false
+  /-- Print the type of a declaration. -/
+  showTypes : Bool := true
+  /--
+  Include internal declarations (names starting with `_`, `match_` or `proof_`)
+  -/
+  internals : Bool := false
+
+/-- Function elaborating `Config`. -/
+declare_config_elab elabPrintPrefixConfig PrintPrefixConfig
 
 /--
 Command for #print prefix
 -/
-syntax (name := printPrefix) "#print prefix " ident : command
+syntax (name := printPrefix) "#print prefix " (Lean.Parser.Tactic.config)? ident : command
+
+/--
+Returns true if any part of name starts with underscore or uses a num.
+
+This can be used to hide internally generated names.
+-/
+def isInternalDetail : Name → Bool
```

Comment:
I think this should be "higher up". If not in core then in `Std.Lean`?

Reply:
I've added it to `Lean.Name` in `Std/Data/Lean` for now   I made [an issue in core](https://github.com/leanprover/lean4/issues/2850), but it didn't come to a clear resolution.

I've put this upstream a bit in `Std` and if other tactics adopt `isInternalDetail` predicate then it would be easy to upstream.

Reply:
Sounds like a good temporary fix to me!